lib/kvdb/kvdb_ctxn.c

```diff
@@ -78,11 +78,12 @@ struct kvdb_ctxn_set_impl {
 static inline void
 kvdb_ctxn_bind_putref(struct kvdb_ctxn_bind *bind)
 {
-    if (atomic_dec_return(&bind->b_ref) == 0) {
-        if (bind->b_ctxn)
-            kvdb_ctxn_h2r(bind->b_ctxn)->ctxn_bind = 0;
-        free(bind);
-    }
+    int refcnt = atomic_dec_return(&bind->b_ref);
```

Comment:
Weird that reference count is signed. Can it ever be negative?

Reply:
Well, signed ref counts are quite a bit easier to debug than unsigned..

Reply:
What prevents some other thread from calling getref() simultaneously with some thread calling putref() which drops the count to zero?

Reply:
Not unless there's an error.